### PR TITLE
fix backward link to deleted key in provenance

### PR DESCRIPTION
**Pull request by cendhu**
When the key is deleted from the world state in
a one block and the same key is added in the future
block, there should be two links (previous/next) between old deleted
value and the new value. Due to a mistake in the unit-test,
we missed to detect the bug in the production code.

Now with the integration test, we could detect this bug.
To get the last deleted value to construct the next/previous
link with the new value, we need to pass the correct key. The
existing code passes the incorrect key to find the last deleted
key node.

### CODE REVIEW

File: internal/provenance/commit_and_query_test.go [hunk 1]
```diff
@@ -847,109 +837,109 @@ func TestGetNextValues(t *testing.T) {
...
-			dbName: "db1",
-			key:    "key3",
-			version: &types.Version{
-				BlockNum: 3,
-				TxNum:    0,
-			},
-			limit:          -1,
-			expectedValues: nil,
-		},
+		// {
```
Comment (tock-ibm) [inline internal/provenance/commit_and_query_test.go:840]: all the tests are commented out...

File: internal/provenance/commit_and_query.go [hunk 2]
```diff
@@ -308,6 +308,7 @@ func (s *Store) getDeletedValuesWithoutLock(dbName, key string) ([]*types.ValueW
 	s.logger.Debugf("fetch all historical deleted values associated with the key [%s] in db [%s]", key, dbName)
 	cKey := constructCompositeKey(dbName, key)
 	p := cayley.StartPath(s.cayleyGraph, quad.String(cKey)).Out().Tag("deleted_value").In(quad.String(DELETES)).Back("deleted_value")
+	// p := cayley.StartPath(s.cayleyGraph, quad.String(cKey)).Out().In(quad.String(DELETES))
```
Comment (tock-ibm) [inline internal/provenance/commit_and_query.go:311]: what is this?

File: internal/provenance/commit_and_query_test.go (hunk 1 above)
Comment (cendhu) [reply internal/provenance/commit_and_query_test.go:840]: Enabled them.

File: internal/provenance/commit_and_query.go (hunk 2 above)
Comment (cendhu) [reply internal/provenance/commit_and_query.go:311]: Forgot to remove

**Comment by tock-ibm** [comment]
```
block5TxsData := []*TxDataForProvenance{
		{
			IsValid: true,
			DBName:  "db1",
			UserID:  "user2",
			TxID:    "tx6",
			Writes: []*types.KVWithMetadata{
				{
					Key:   "key1",
					Value: []byte("value5"),
					Metadata: &types.Metadata{
						Version: &types.Version{
							**BlockNum: 4,**            //<<< should be 5
							TxNum:    0,
						},
					},
				},
			},
              }
```
and
```
	block6TxsData := []*TxDataForProvenance{
		{
			IsValid: true,
			DBName:  "db1",
			UserID:  "user2",
			TxID:    "tx50",
			Deletes: map[string]*types.Version{
				"key1": {
					**BlockNum: 4,**        //<<< should be 5 as well
					TxNum:    0,
				},
			},
		},
	}
```

**Comment by cendhu** [comment]
Done

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**